### DISCUSSION TITLE
openapi 3 for rdf4j-server?

### DISCUSSION THREAD

**Discussion by seralf** [eclipse-rdf4j/rdf4j#4269] [Ideas]
Hi, 
I wonder if there are plans to include a full support of openapi for the server?

This may help anyone who has needs for testing and exploring the API, as well as some programmers who needs to write their own clients, and possibly generating a stub, starting from the usual openapi-based generators.

Starting from that considerations, and looking at the available version, as a static file:
https://github.com/eclipse/rdf4j/blob/main/tools/server/src/main/webapp/swagger.yaml

I tried to use the [REST API documentation](https://rdf4j.org/documentation/reference/rest-api/), in order to add the missing endpoints descriptions. I upload [here](https://github.com/eclipse/rdf4j/files/9932695/rdf4j-server_openapi.yaml.zip) the updated draft: it is still missing some elment, and there could be errors. The idea is to fix it, than convert it to a proper openapi version 3 format, and it may be useful for others.

What do you think about?
Alfredo

**Comment by barthanssens** [comment]
Well... it sure is worth the try.

IMHO it would indeed be useful for documentation purposes, though I'm not sure if the stub-generation would really save that much work  (wouldn't a programmer just start with a SPARQL client anyway, and add some bits for listing / creating repos manually ?)
